src/error.rs

```diff
@@ -6,4 +7,26 @@ pub enum Error {
     Io(#[from] std::io::Error),
     #[error("Serde: {0}")]
     Serde(#[from] serde_json::Error),
+    #[error("Reqwest: {0}")]
+    Reqwest(#[from] reqwest::Error),
+    #[error("Oneshot: {0}")]
+    Oneshot(#[from] tokio::sync::oneshot::error::RecvError),
+    #[error("Download: {0}")]
```

Comment:
Maybe, not sure, `RecvError` could map to a more general channel cancellation variant instead of adding a dedicated Oneshot error